native/src/widget/slider.rs

```diff
@@ -152,15 +152,22 @@ where
     ) {
         let mut change = || {
             let bounds = layout.bounds();
-
             if cursor_position.x <= bounds.x {
                 messages.push((self.on_change)(*self.range.start()));
             } else if cursor_position.x >= bounds.x + bounds.width {
                 messages.push((self.on_change)(*self.range.end()));
             } else {
+                let step_magnitude = self.step.log10().round() as i32;
+                let scaling = 10.0f32.powi(-step_magnitude);
```

Comment:
The purpose of this stuff is to be able to round `value` to the same number of digits that `step` has. Without it, the color sliders in the `tour` example (which have values in [0,1] with step size 0.01) would sometimes display values like 0.149999, which should of course be rounded to 0.15. I hope making the precision of value dependent on step is ok.

Reply:
As far as I understand, this is a floating point precision issue. In that case, dividing by the `scaling` could also cause the result to be inaccurate, right?

If that is the case, then I believe we should simply compute `steps * self.step + self.range.start()` in `f64` and then convert it back to `f32`. We could also consider switching to `f64` for the `Slider` API.

Reply:
Simply doing this
```rust
let percent = (cursor_position.x - bounds.x) / bounds.width;
let steps = (percent * (self.range.end() - self.range.start())
    / self.step)
    .round();
let value = (steps * self.step + self.range.start()) as f64;
messages.push((self.on_change)(value as f32));
```
does not fix the problem. Is this what you meant by computing compute `steps * self.step + self.range.start()` in `f64`, or did you have something more involved in mind?

What do you think about treating this as a display problem and letting the `Renderer` take care of it?

Reply:
I meant computing the value using double precision:

```rust
let value = f64::from(steps) * f64::from(self.step) + f64::from(self.range.start());
```

Reply:
Ah, I didn’t know about this. I’ll give it a try.

Reply:
I just tried it, it didn't fix the issue.

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f91a49afbb7e882a7d6ab356ab018b5b